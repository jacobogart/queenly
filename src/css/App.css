.App {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 70%;
  margin: 0 auto;
  position: fixed;
  width: 100%;
  z-index: 1;
}

.backgroundImage {
  height: 100vh;
  left: 0;
  margin: 0 auto;
  position: fixed;
  right: 0;
  width: 100vw;
  /* background: url("https://images.unsplash.com/photo-1512073490563-2fde3bf833ad?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=2850&q=80"); */
  background: url("https://images.unsplash.com/photo-1552297777-dacae7d2b894?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80");
  background-position: center 25%;
  filter: brightness(0.5) sepia(30%) hue-rotate(270deg);
  z-index: 0;
}
.mainContent {
  align-items: center;
  border: 1px solid red;
  display: flex;
  height: 300px;
  justify-content: center;
  text-align: center;
  width: 500px;
}

.navBar {
  background-color: rgb(32, 39, 39);
}
