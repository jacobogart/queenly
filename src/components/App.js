import React from "react";
import NavBar from "./NavBar";
import search from "../helpers.js";
import SplashPage from "./SplashPage";
import Card from "./Card";
import SearchResults from "./SearchResults";

// * CSS imports
import "../css/App.css";

class App extends React.Component {
  constructor() {
    super();
    this.state = {
      searchResults: [],
      currentResult: null,
      showCard: false,
      showResults: false,
      showSplash: true,
      cardComponent: null,
      showSuggestions: false,
      bars: [{shows: []}],
      queens: []
    };
  }

  componentDidMount() {
    fetch(
      "https://fe-apps.herokuapp.com/api/v1/whateverly/1901/jacobogart/bars"
    )
      .then(response => response.json())
      .then(data => this.setState({ bars: data.bars }))
      .catch(err => console.log(err));

    fetch(
      "https://fe-apps.herokuapp.com/api/v1/whateverly/1901/jacobogart/queens"
    )
      .then(response => response.json())
      .then(data => this.setState({ queens: data.queens }))
      .catch(err => console.log(err));
  }

  updateResults = query => {
    this.setState({
      showSuggestions: true,
      searchResults: search(query, this.state.bars, this.state.queens)
    });
  }

  selectResult = resultName => {
    if (this.state.showSplash || this.state.showResults) {
      this.toggleCard();
    }

    this.setState({
      currentResult: resultName,
      showSplash: false,
      cardData: search(resultName, this.state.bars, this.state.queens)[0],
      showSuggestions: false
    });
  }

  toggleCard = () => {
    this.setState({
      showCard: true,
      showResults: false,
      showSplash: false
    });
  }

  toggleResults = () => {
    this.setState({
      showCard: false,
      showResults: true,
      showSplash: false
    });
  }

  toggleSplash = () => {
    this.setState({
      showCard: false,
      showResults: false,
      showSplash: true
    });
  }

  render() {
    let card;

    let cardComponent = 
      <Card
        cardData={this.state.cardData}
        toggleCard={this.toggleCard}
        toggleSplash={this.toggleSplash}
        bars={this.state.bars}
        queens={this.state.queens}
      />

    let searchResultsComponent = (
      <SearchResults
        searchResults={this.state.searchResults}
        toggleResults={this.toggleResults}
        selectResult={this.selectResult}
      />
    );

    let splashPageComponent =
      <SplashPage
        toggleResults={this.toggleResults}
        updateResults={this.updateResults}
        selectResult={this.selectResult}
        searchResults={this.state.searchResults}
        showSuggestions={this.state.showSuggestions}
      />

    if (this.state.showSplash) {
      card = splashPageComponent
    } else if (this.state.showResults) {
      card = searchResultsComponent
    } else {
      card = cardComponent
    }

    return (
      <div>
        <header className="navBar">
          <NavBar
            searchBarDisplay={this.state.showCard}
            updateResults={this.updateResults}
            selectResult={this.selectResult}
            searchResults={this.state.searchResults}
            showSuggestions={this.state.showSuggestions}
          />
        </header>
        <section className="App">
          <article className="mainContent">
            {card}
          </article>
        </section>
        <div className="appBackground" />
      </div>
    );
  }
}

export default App;
