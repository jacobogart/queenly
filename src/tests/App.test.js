import React from 'react';
import ReactDOM from 'react-dom';
import App from '../components/App.js';
import { shallow } from 'enzyme';

// const [nameOfFunctionHere] = jest.fn();

describe('App', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should match snapshop', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      searchResults: [],
      showCard: false,
      currentResult: null
    });
  });

  it('should setState, searchResults, to an array of names', () => {
    expect(wrapper.state('searchResults')).toEqual([]);
// todo: figure out how to simulate this test
    // const query = {
    //   target: [`Roscoe's Tavern`, `Roscoe's Bar`]
    // };

    // wrapper.instance().updateResults(query);
    // expect(wrapper.state('searchResults').length).toEqual(query.length);
  });

  it('should define current result when selectResult is invoked, and toggle card', () => {
    expect(wrapper.state('showCard')).toEqual(false);
    expect(wrapper.state('currentResult')).toEqual(null);

    const resultName = {
      target: `Roscoe's Tavern`
    };

    wrapper.instance().selectResult(resultName);
    expect(wrapper.state('showCard')).toEqual(true);
    expect(wrapper.state('currentResult')).toEqual(resultName);
  });
});