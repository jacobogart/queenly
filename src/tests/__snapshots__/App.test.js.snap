// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshop 1`] = `
<div>
  <header
    className="navBar"
  >
    <NavBar
      displayAllOfType={[Function]}
      displayAllSearchResults={[Function]}
      displaySplashPage={[Function]}
      hideSearchSuggestions={[Function]}
      searchBarDisplay={false}
      searchResults={Array []}
      selectSearchResult={[Function]}
      showSplashPage={true}
      updateQuery={[Function]}
      updateSearchResults={[Function]}
    />
  </header>
  <section
    className="App"
  >
    <SplashPage
      displayAllSearchResults={[Function]}
      hideSearchSuggestions={[Function]}
      searchResults={Array []}
      selectSearchResult={[Function]}
      updateQuery={[Function]}
      updateSearchResults={[Function]}
    />
  </section>
  <div
    className="backgroundImage"
  />
</div>
`;
